paddle/fluid/operators/elementwise/elementwise_pow_op.h

```diff
@@ -12,17 +12,59 @@ limitations under the License. */
 #pragma once
 
 #include <cmath>
+#include <type_traits>
 #include "paddle/fluid/operators/elementwise/elementwise_op.h"
 #include "paddle/fluid/operators/elementwise/elementwise_op_function.h"
 
 namespace paddle {
 namespace operators {
 
-template <typename T>
-struct PowFunctor {
+namespace details {
+
+// Only valid if T is an integral type.
+template <class T,
+          class = typename std::enable_if<std::is_integral<T>::value>::type>
+static inline HOSTDEVICE T ipow(T base, T exp) {
+  if (exp < 0) {
+    // Note(zhiqiu): In some library, like numpy and pytorch,
+    // integers to negative integer powers are not allowed.
+    // Since a^b = 1/(a^(-b)) when b < 0, it will truncate to 0.
+    // Return 0, for specialization to avoiding infinite loop.
+    return 0;
+  }
+
+  T result = 1;
```

Comment:
Add a link where the following codes from?

Reply:
Thanks, implementation changed.